flink-ml-lib/src/main/java/org/apache/flink/ml/clustering/kmeans/KMeans.java

```diff
@@ -254,58 +272,150 @@ public Tuple3<Integer, DenseVector, Long> map(Tuple2<Integer, DenseVector> value
                             DenseVector, DenseVector[], Tuple2<Integer, DenseVector>>,
                     IterationListener<Tuple2<Integer, DenseVector>> {
         private final DistanceMeasure distanceMeasure;
-        private ListState<DenseVector> points;
-        private ListState<DenseVector[]> centroids;
+        private ListState<DenseVector[]> centroidsState;
+        private DenseVector[] centroids;
```

Comment:
Does this improve performance by using `centroids`? If not, it seems simpler to not adding this variable.